src/material-experimental/mdc-slider/slider.ts

```diff
@@ -48,6 +56,194 @@ export interface MatSliderDragEvent {
   value: number;
 }
 
+/**
+ * The visual slider thumb.
+ *
+ * Handles the slider thumb ripple states (hover, focus, and active),
+ * and displaying the value tooltip on discrete sliders.
+ */
+@Component({
+  selector: 'mat-slider-visual-start-thumb, mat-slider-visual-end-thumb',
+  templateUrl: './slider-thumb.html',
+  styleUrls: ['slider-thumb.css'],
+  host: {
+    'class': 'mdc-slider__thumb',
+  },
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+})
+export class MatSliderVisualThumb implements AfterViewInit, OnDestroy {
+  /** Whether the slider displays a numeric value label upon pressing the thumb. */
+  @Input() _discrete: boolean;
```

Comment:
This who class is internal, so just put `@docs-private` in the class doc string and don't bother with underscores for the inputs